### ISSUE TITLE
Silently shifting schedules

### ISSUE THREAD

**Issue by faqtos** [orgzly/orgzly-android#655]
- [x] I have searched for existing issues that may be the same as or related to mine.

I have created an item scheduled for Jan 10, with a yearly recursion, say a birthday reminder.
I didn't act on this item at the scheduled date. On Jan 13 I look at the Agenda widget and there it appears as if the item was scheduled for today, with no hint that it is actually 3 days late. That is a first problem, overdue items should probably appear at the top of the list, showing them as scheduled for today is showing erroneous info.

But the worst is that this seems not just to be a problem linked to the representation of search results: I realized that the scheduled time of the item was actually silently modified, so for upcoming years it is now scheduled on Jan 13 every year. Something I only caught because I manually went to check if the + 1 year thing had worked correctly, else my agenda would have stayed corrupted without me realizing it.

**Comment by nevenz** [comment]
> overdue items should probably appear at the top of the list, showing them as scheduled for today is showing erroneous info.

Some indicator is definitely needed.

> I realized that the scheduled time of the item was actually silently modified, so for upcoming years it is now scheduled on Jan 13 every year. Something I only caught because I manually went to check if the + 1 year thing had worked correctly, else my agenda would have stayed corrupted without me realizing it.

Which repeater type did you use? `++1y` would be the one to use here. Although there's no automatic time shifting anyway (yet, there's #397, though I'm not really a fan of doing that).

Is it possible you used `.+` or `+` and marked the note as done on 12th?

**Comment by nevenz** [comment]
> I see commit 2e4a810 added an "Overdue" heading to the agenda. How can this be deactivated and the items put at the start or end of today's agenda, please? Having to wade through an "Overdue" section before seeing today's timely tasks is rather demotivational.

Wouldn't you still have to go through them if they are at the beginning of today?

Perhaps adding up/down arrows to jump to the previous or next section might be useful anyway? Either on each divider or in the action bar.

**Comment by mjray** [comment]
> Wouldn't you still have to go through them if they are at the beginning of today?

Yes, but that still removes the demotivational "Overdue" header.

Up/down arrows might be useful anyway, yes.
